src/ape/managers/project.py

```diff
@@ -125,14 +134,17 @@ def file_needs_compiling(source: Path) -> bool:
         needs_compiling = list(filter(file_needs_compiling, sources))
 
         # Set the context in case compiling a dependency (or anything outside the root project).
-        with self.compilers.config.using_project(self.path, contracts_folder=self.contracts_folder):
-            self.compilers.config.PROJECT_FOLDER = self.path
-            self.compilers.config.contracts_folder = self.contracts_folder
-            compiled_contract_types = self.compilers.compile(needs_compiling)
+        with self.compiler_manager.config_manager.using_project(
+            self.path, contracts_folder=self.contracts_folder
+        ):
+            self.compiler_manager.config_manager.PROJECT_FOLDER = self.path
```

Comment:
```suggestion
```

This line and the one below happen already in the `using_project()` method, so I am not sure they are still needed here